punchcut/Cargo.toml

```diff
@@ -0,0 +1,18 @@
+[package]
+name = "punchcut"
+version = "0.1.0"
+edition = "2021"
+license = "MIT/Apache-2.0"
+description = "Loading glyphs from OpenType font files."
+repository = "https://github.com/googlefonts/fontations"
+readme = "README.md"
+categories = ["text-processing", "parsing", "graphics"]
+
+[dependencies]
+read-fonts = { version = "0.0.4", path = "../read-fonts" }
+peniko = { git = "https://github.com/linebender/peniko" }
```

Comment:
If I understand this correctly, this dependency is used for the `Shape` impl on top of `Outline`. Could this dependency be made optional, and/or the `Shape` implementation be a separate pull request? If/when punchcut is used in a Skia backend to extract outlines, I think we could avoid the peniko/kurbo dependency.

Reply:
We could remove it at this time, but the dependency exists for future COLRv1 support where we'll need to represent affine transforms and the various brush types. We don't absolutely require it... we can define our own types in this crate, but I'd prefer to keep it so that those types are consistent across the ecosystem. Perhaps we can do some measurement on binary size to see what the actual effect is?

Reply:
If I am doing this right (see below) looks like `kurbo` is 1.8M and `peniko` 459k. IMO this is quite substantial. In Chrome we already get warnings for Android binary size (which we need to keep small for being able to install Chrome effectively through play store) for CLs that increase the binary size by 20k. 

As for the initial prototype we probably won't have COLRv1 support through punchcut immediately, I'd still prefer to make this dependency optional and include it, ideally optionally too, later for COLRv1 support in punchcut.

> represent affine transforms and the various brush types.

By "various brush types" do you mean the gradient definitions? By brush I tend to imagine something like a path-effect or shape to a line or curve stroke. If we're only interested in the type definitions for the interface, it seems quite heavy to depend on all of peniko and kurbo if the size impact is the whole library - but I am not sure to what extent the linking steps can prune unused parts.

`$ cargo build -r --lib`

then 

```bash
~/dev/peniko/target/release/deps$ ls -lha *rlib
-rw-r--r-- 1 drott primarygroup 502K Jan 12 10:37 libarrayvec-b8ff81c131b08d72.rlib
-rw-r--r-- 1 drott primarygroup 1.8M Jan 12 10:37 libkurbo-e1fffba60c3a2368.rlib
-rw-r--r-- 2 drott primarygroup 459K Jan 12 10:37 libpeniko-4b15967201e87c2a.rlib
-rw-r--r-- 1 drott primarygroup 444K Jan 12 10:37 libsmallvec-cd2b8e69191b620b.rlib
```

Reply:
Thanks for the analysis. We’ve settled on limiting dependencies to crates in this repo so I’ll remove these and perhaps restore them later behind a feature gate.

I wanted to note that rlibs contain full binary code along with a significant chunk of metadata and IR to facilitate incremental builds and generic instantiation in downstream crates. They’re not particularly representative of final artifact size.

Reply:
+1 to gate this on removal of those deps. Also ... how does one correctly measure binary size?

Reply:
Filed https://github.com/googlefonts/fontations/issues/200 to establish an accepted and easily repeated mechanism to measure size impact.